test/functional/shell/viml_system_spec.lua

```diff
@@ -200,6 +201,27 @@ describe('system()', function()
       eq('* $NOTHING ~/file',
          eval("system(['echo', '-n', '*', '$NOTHING', '~/file'])"))
     end)
+
+    it('quotes arguments correctly', function()
+      local pwd = helpers.funcs.getcwd()
+      local fname
+      if ffi.os == "Windows" then
+        fname = "countargs.bat"
+      else
+        fname = "countargs.sh"
+      end
+      local script = pwd .. '/test/functional/shell/' .. fname
+      local out0 = eval('system("' .. script .. ' 1 \\\"2 3\\\"")')
+
+      local cmd1
+      if ffi.os == "Windows" then
+        cmd1 = 'system(["cmd", "/c", "' .. script ..'", "1", "2 3"])'
+      else
+        cmd1 = 'system(["' .. script ..'", "1", "2 3"])'
```

Comment:
To be equivalent with the `cmd /c` invocation above, shouldn't this be:

    cmd1 = 'system(["sh", "-c", "' .. script ..'", "1", "2 3"])'